src/main/java/seedu/address/model/session/RecurringSession.java

```diff
@@ -78,6 +85,15 @@ public boolean hasSessionOnDate(SessionDate sessionDate) {
         return !endBefore(sessionDate) && isConsistentDatesAndInterval(getSessionDate(), sessionDate, getInterval());
     }
 
+    /**
+     * Builds a session with the same time but with the specified time.
+     */
+    public Session buildSessionOnDate(LocalDate date) {
+        requireNonNull(date);
+        SessionDate sessionDate = new SessionDate(date.toString(), getSessionDate().getTime().toString());
```

Comment:
This looks like the same thing to me as onSessionDate (but without the assert), with different input.
Perhaps rename both methods to reflect this difference in input.
And for the former, call onSessionDate(new SessionDate()) from within

I like you use the word build better than just 'on'

Reply:
actually right, i tried to use your onSessionDate() method but because I needed to pass SessionDate in as input, i found it very troublesome to use it. My this build method has the added benefit that it uses the time info of the recurring session, so i don't have to retrieve it, create a new SessionDate and then pass it into the onSessionDate() method.

Reply:
yeah im suggesting you overload the method to allow localDateTime input

Reply:
i.e. 
1. rename onSessionDate(SessionDate sd) method to buildOnSessionDate(SessionDate sd)
2. make a 
public Session buildOnSessionDate(LocalDate date) {
SessionDate sessionDate = new SessionDate ......
return onSessionDate(sessionDate);
}

Reply:
ohh ok, that's a good idea! I will change it to this.